pkgs/merkle-tree-worker/src/worker.ts

```diff
@@ -4,8 +4,19 @@ import { IMerkleTreeWasm, IMerkleTreeWorker } from './interface'
 let merkleTreeWasm: IMerkleTreeWasm
 
 export const merkleTreeWorker: IMerkleTreeWorker = {
-  generateMerkleProof(leaves, leaf, depth): Uint8Array {
-    return merkleTreeWasm.generate_merkle_proof(leaves, leaf, depth)
+  async generateMerkleProof(leaves, leaf, depth): Promise<Uint8Array> {
+    // eslint-disable-next-line no-useless-catch
+    try {
+      // eslint-disable-next-line  @typescript-eslint/await-thenable
+      const result = await merkleTreeWasm.generate_merkle_proof(
+        leaves,
+        leaf,
+        depth,
+      )
+      return result
+    } catch (error) {
+      throw error
+    }
```

Comment:
No need for the try/catch block if we just rethrow the error as is.

```suggestion
return merkleTreeWasm.generate_merkle_proof(leaves, leaf, depth)
```